trunner/testcase.py

```diff
@@ -126,6 +136,13 @@ def handle_pyexpect_error(self, proc, exc):
         self.exception += Color.colorify('GOT:\n', Color.BOLD)
         self.exception += f'{got}\n'
 
+    def handle_unicode_error(self, proc):
```

Comment:
I see no separation between private and public class members here. Just pointing that as I hardly see any underscore prefixed functions in trunner so this is out of scope of this review.

Reply:
I've created issue for this: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/331. I will think about it in meantime.